### ISSUE TITLE
bad result on custom colmap data

### ISSUE THREAD

**Issue by huang997733** [bennyguo/instant-nsr-pl#56]
Hi, 

I found that the result of neus-colmap is really bad. I used colmap on nerf_synthetic data then trained the model, and the result is very bad.

- colmap on nerf_synthetic/chair, radius: 1
<img width="1504" alt="image" src="https://user-images.githubusercontent.com/88305416/228139914-c2aebb32-477d-4dd9-a987-bebcddb8792d.png">

I have also tried to change radius in neus-colmap.yaml file as in #20. I have tried 0.5, 1, 1.5 which all gave me bad results. 

- colmap on nerf_synthetic/chair, radius: 0.5
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/88305416/228140422-78793ef9-7a6d-4c89-b77a-465af06f167b.png">

- colmap on nerf_synthetic/chair, radius: 1.5
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/88305416/228140476-bae5a158-a4bb-4640-b831-0fdab944ba17.png">

I tested nerf-colmap, which seems to be working, but neus-colmap isn't working on the nerf_synthetic data. Any idea what went wrong here? 

Thanks

**Comment by bennyguo** [comment]
Could you please make sure that `dataset.use_mask` is set to false and  `system.loss.lambda_mask` is set to 0? If `dataset.use_mask=true` and you don't provide a `masks` folder, the object masks loaded would be all white... I'll change this behavior soon.

**Comment by huang997733** [comment]
I set dataset.use_mask to false and system.loss.lambda_mask to 0, still giving me bad results. Any other ideas on what could go wrong here?

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/88305416/228728090-6abebe33-febf-4bd7-a385-6c0e5ce9fc7d.png">

**Comment by huang997733** [comment]
Thanks a lot!
There is a cube when using neus-colmap.yaml, still something wrong here.

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/88305416/229044735-e547c315-fed6-4339-8b9c-6987f259bb31.png">

**Comment by huang997733** [comment]
Thanks a lot!

**Comment by TianyuHuang-000** [comment]
> I just changed the logic about using masks. If you are training on NeRF-Synthetic data with COLMAP poses, you should use `configs/neus-colmap.yaml` and do the following:
> 
> * extract masks from images and store them in a `masks` folder alongside with `images`
> * set `system.loss.lambda_mask=0.1`
> * set `model.radius=0.5`

Thanks for the update
Does it mean that the colmap data has to be used with a mask folder as compulsory element?
Before I can run neus_colmap on the "room" scene from mip360 dataset, but now it doesn't work.
Some scenario may target to reconstruct the background surface as well and hard to give a mask, any work could be done on that?

**Comment by bennyguo** [comment]
No, you don't need to train COLMAP data with masks. Actually, the main point for implementing a background model is to enable training without masks on real-captured scenes.

Which version did you use to train on the room scene before? Did you enlarge `model.radius`? It is a special case as no foreground object is presented in the scene. I can get reasonable results by setting `model.radius=5` (a large value) in the latest code.

**Comment by A-pril** [comment]
I meet this problem too. I only can use nerf-colmap, the test result is very bad and I can't get a obj file(the obj file is 0 KB).The test scene is a room , and I take photos by myself.  I don't have masks. I have tried radius with 1 and 0.5 and neither is good. test/psnr is 5-7.

- The very bad result

![image](https://user-images.githubusercontent.com/76250553/230395287-b592fc87-11b1-45d9-8494-91a9fb63e3ea.png)

- The relative good result

![image](https://user-images.githubusercontent.com/76250553/230395383-414be035-1043-4f1d-b6ac-175ee3140a30.png)

Do u have any advice on this issue? maybe the param values is not correct?

**Comment by bennyguo** [comment]
@A-pril Could you try the latest code (with a bugfix on camera normalization)? See https://github.com/bennyguo/instant-nsr-pl/issues/55 for more discussions.
